$OpenBSD: patch-gnuradio-runtime_lib_buffer_cc,v 1.1 2021/09/05 08:13:33 ajacoutot Exp $

[PATCH] boost: remove deprecated math/common_factor.hpp

2c767bb260a25b415e8c9c4b3ea37280b2127cec

Index: gnuradio-runtime/lib/buffer.cc
--- gnuradio-runtime/lib/buffer.cc.orig
+++ gnuradio-runtime/lib/buffer.cc
@@ -25,22 +25,13 @@
 #endif
 #include "vmcircbuf.h"
 #include <gnuradio/buffer.h>
+#include <gnuradio/integer_math.h>
 #include <gnuradio/math.h>
 #include <assert.h>
 #include <algorithm>
 #include <iostream>
 #include <stdexcept>
 
-// the following header is deprecated as of Boost 1.66.0, and the
-// other API was introduced in Boost 1.58.0. Since we still support
-// Boost back to 1.54.0, use the older API if pre-1.5.80 and otherwise
-// use the newer API.
-#if (BOOST_VERSION < 105800)
-#include <boost/math/common_factor_rt.hpp>
-#else
-#include <boost/integer/common_factor_rt.hpp>
-#endif
-
 namespace gr {
 
 static long s_buffer_count = 0; // counts for debugging storage mgmt
@@ -80,13 +71,9 @@ static long s_buffer_reader_count = 0;
  *
  *     type_size * nitems == k * page_size
  */
-static long minimum_buffer_items(long type_size, long page_size)
+static inline long minimum_buffer_items(long type_size, long page_size)
 {
-#if (BOOST_VERSION < 105800)
-    return page_size / boost::math::gcd(type_size, page_size);
-#else
-    return page_size / boost::integer::gcd(type_size, page_size);
-#endif
+    return page_size / GR_GCD(type_size, page_size);
 }
 
 
